modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/checkout/util/test/BillingAddressCommerceCheckoutStepTest.java

```diff
@@ -0,0 +1,210 @@
+
+		Country country = CommerceInventoryTestUtil.addCountry(serviceContext);
+
+		Address address = _addressLocalService.addAddress(
+			RandomTestUtil.randomString(), _user.getUserId(),
+			AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
+			country.getCountryId(), 0, 0, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), false, RandomTestUtil.randomString(),
+			true, RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), null, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), serviceContext);
+
+		CommerceCurrency commerceCurrency =
+			CommerceCurrencyTestUtil.addCommerceCurrency(_group.getCompanyId());
+
+		CommerceChannel commerceChannel = CommerceTestUtil.addCommerceChannel(
+			_group.getGroupId(), commerceCurrency.getCode());
+
+		_commerceChannelAccountEntryRelLocalService.
+			addCommerceChannelAccountEntryRel(
+				_user.getUserId(), accountEntry.getAccountEntryId(),
+				Address.class.getName(), address.getAddressId(),
+				commerceChannel.getCommerceChannelId(), true, 0,
+				CommerceChannelAccountEntryRelConstants.TYPE_BILLING_ADDRESS);
+
+		_commerceOrder = _commerceOrderLocalService.addCommerceOrder(
+			_user.getUserId(), commerceChannel.getGroupId(),
+			accountEntry.getAccountEntryId(), commerceCurrency.getCode(), 0);
+
+		_commerceOrder.setShippingAddressId(address.getAddressId());
+
+		_commerceOrder = _commerceOrderLocalService.updateCommerceOrder(
+			_commerceOrder);
+	}
+
+	@Test
+	public void testBillingAddressIsVisibleToBuyer() throws Exception {
+		Assert.assertTrue(_isBillingAddressCommerceCheckoutStepActive());
+
+		_resourcePermissionLocalService.removeResourcePermission(
+			_group.getCompanyId(), CommerceOrderConstants.RESOURCE_NAME,
+			ResourceConstants.SCOPE_GROUP_TEMPLATE,
+			String.valueOf(GroupConstants.DEFAULT_PARENT_GROUP_ID),
+			_role.getRoleId(), CommerceOrderActionKeys.VIEW_BILLING_ADDRESS);
+
+		try {
+			Assert.assertFalse(_isBillingAddressCommerceCheckoutStepActive());
+		}
+		finally {
+			_resourcePermissionLocalService.addResourcePermission(
+				_group.getCompanyId(), CommerceOrderConstants.RESOURCE_NAME,
+				ResourceConstants.SCOPE_GROUP_TEMPLATE,
+				String.valueOf(GroupConstants.DEFAULT_PARENT_GROUP_ID),
+				_role.getRoleId(),
+				CommerceOrderActionKeys.VIEW_BILLING_ADDRESS);
+		}
+	}
+
+	private boolean _isBillingAddressCommerceCheckoutStepActive()
+		throws Exception {
+
+		HttpServletRequest httpServletRequest = new MockHttpServletRequest();
+
+		httpServletRequest.setAttribute(
+			CommerceCheckoutWebKeys.COMMERCE_ORDER, _commerceOrder);
+
+		ThemeDisplay themeDisplay = ThemeDisplayFactory.create();
+
+		themeDisplay.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(_user));
+		themeDisplay.setScopeGroupId(_group.getGroupId());
+		themeDisplay.setSignedIn(true);
+		themeDisplay.setUser(_user);
+
+		httpServletRequest.setAttribute(WebKeys.THEME_DISPLAY, themeDisplay);
```

Comment:
what about if we extract this in a method `_getMockHttpServletRequest` and we remove this `_isBillingAddressCommerceCheckoutStepActive` and in the test we just put

Assert.assertTrue(_commerceCheckoutStep.isActive(_getMockHttpServletRequest(), null));